third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -634,6 +634,53 @@ class ConvertAddPtrOp : public PointerCanonicalizationPattern<tt::AddPtrOp> {
   }
 };
 
+class ConvertExtractSliceOp
```

Comment:
can you add a little short doc string here for the canonicalization - i believe it's somethine like

```
slice(fatPtrBase, fatPtrOffset) -> (fatPtrBase, slice(fatPtrOffset)),
```

(which btw makes sense to me).